src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -23,13 +23,13 @@ internal class NoOpRestoreUtilities
         /// </summary>
         internal static bool IsNoOpSupported(RestoreRequest request)
         {
-            return request.DependencyGraphSpec != null && request.ProjectStyle != ProjectStyle.DotnetCliTool;
+            return request.DependencyGraphSpec != null;
         }
 
         /// <summary>
         /// The cache file path is $(BaseIntermediateOutputPath)\$(project).nuget.cache
         /// </summary>
-        internal static string GetPJorPRCacheFilePath(RestoreRequest request)
+        private static string GetPJorPRCacheFilePath(RestoreRequest request)
```

Comment:
a bettern name could be GetBuildIntegratedProjectCacheFile, and might be better if you store it as a field on the base class of those kinds of projects.